src/crimson/osd/osd_operations/background_recovery.cc

```diff
@@ -164,11 +172,17 @@ PglogBasedRecovery::do_recovery()
   if (pg->has_reset_since(epoch_started)) {
     return seastar::make_ready_future<bool>(false);
   }
-  return with_blocking_event<RecoveryBackend::RecoveryBlockingEvent,
-			     interruptor>([this] (auto&& trigger) {
-    return pg->get_recovery_handler()->start_recovery_ops(
-      trigger,
-      crimson::common::local_conf()->osd_recovery_max_single_start);
+  auto fut = interruptor::now();
+  if (pg->have_unfound()) {
+    fut = pg->find_unfound(epoch_started);
```

Comment:
Let's just modify find_unfound to return a ready future if there are no unfound.  That'll simplify all of the callers.

Reply:
Sorry for the delay, I forgot to address this comment, please take a look again, thanks.